polytracker/mapping.py

```diff
@@ -4,24 +4,59 @@
 
 from collections import defaultdict
 from pathlib import Path
-from typing import Dict, List, Set, Tuple, Generator
+from typing import Dict, List, Set, Tuple, Iterator
 from tqdm import tqdm
-from mmap import mmap, PROT_READ
 
 from .plugins import Command
-from .taint_dag import TDFile, TDRangeNode, TDSourceNode, TDUnionNode
+from .taint_dag import TDFile, TDNode, TDRangeNode, TDSourceNode, TDUnionNode
 
 
+LabelType = int
```

Comment:
Good move with the types!
I suggest that we take it one step further using https://docs.python.org/3/library/typing.html#newtype

Reply:
This causes a few headaches, like excessive typecasting. For example with `range(n.first, n.last + 1)` when handling `TDRangeNode`. Mypy doesn't recognize `range` as the built-in function and thinks it's a type. Moreover `NewType` is said to introduce runtime overhead. I think we're fine with just the type aliases so far.

Reply:
👍